lib/credit_card_validations/detector.rb

```diff
@@ -28,8 +28,14 @@ def brand(*keys)
     def valid_number?(*keys)
       selected_brands = keys.blank? ? self.brands : resolve_keys(*keys)
       if selected_brands.any?
+        matched_brands = []
         selected_brands.each do |key, brand|
-          return key if matches_brand?(brand)
+          match_data = matches_brand?(brand)
+          matched_brands << {brand: key, matched_prefix_length: match_data.to_s.length} if match_data
+        end
+
+        if matched_brands.present?
+          return matched_brands.sort{|a, b| a[:matched_prefix_length] <=> b[:matched_prefix_length]}.last[:brand]
```

Comment:
@pavelgyravel  i think can be implemented in more easy way , commented PR with separate message take a look please